feat: Modal manager

corn-potage opened the pull request:
- adds modal manager
- modal components were modified to support save, cancel functionality
- decouples open node dir dialog and update node dir
- automatically saves new settings keys with default values from client configs when node settings is opened

known bugs:
- node command not updating after each input change (low priority)

Would like to revisit, to optimize, and clean some of the hacky methods I used in NodeSettingsWrapper, which should also fix the node command update.

jgresham commented on src/main/main.ts:
```diff
@@ -88,13 +88,15 @@ const createWindow = async () => {
     height: 820,
     icon: getAssetPath('icon.png'),
     webPreferences: {
+      enableBlinkFeatures: 'CSSColorSchemeUARendering',
```
Does this fix scroll bars in dark mode?

jgresham commented on src/renderer/Presentational/AddNodeStepper/AddNodeStepperModal.tsx:
```diff
@@ -0,0 +1,109 @@
+// This component could be made into a Generic "FullScreenStepper" component
+// Just make sure to always render each child so that children component state isn't cleard
```
+1 for comments :)

jgresham commented on src/renderer/Presentational/ModalManager/ModalManager.tsx:
```diff
@@ -0,0 +1,54 @@
+import { useSelector } from 'react-redux';
```
Very clean!

jgresham commented on src/renderer/Presentational/ModalManager/modalUtils.tsx:
```diff
@@ -0,0 +1,60 @@
+  updateUnavailable: 'updateUnavailable',
+});
+
+export const modalOnChangeConfig = async (
```
Could you add a function description comment for this? It seems key to adding a new modal. Let's say a description geared for me, if I wanted to add a new modal

jgresham commented on src/renderer/state/modal.ts:
```diff
@@ -0,0 +1,49 @@
+}
+
+export interface ModalConfig {
+  preferences: object;
```
Should these be optional params? thinking modalConfig for nodeSettings wouldn't want to set the other values

corn-potage replied:
Oohh good find, this is some old code and we have ModalConfig interface in modalUtils so this isn't necessary. will delete

jgresham reviewed (commented):
remaining talking points:
- [x] Node requirements back into add node modal style
- [ ] A few comments I left

And:
Since last review the changes to MM and modal make them much more decoupled from the content within the modals and cleaner. And overall, this are massive changes that required knowing how things work from end-to-end. Great job on getting to know how the app works and having a good control over it! Congrats! There is a lot going on!

corn-potage commented:
@jgresham Thanks!! I just added Node Reqs back into modals, and to match the spec provided, I added isDockerRunning check to determine whether to start the node on Node Req page, or move the user to the Docker installation step. Works perfectly when you turn on/off Docker on the 2nd step! Ready to be reviewed again and merged
